src/Nethermind/Nethermind.Blockchain/Nethermind.Blockchain.csproj

```diff
@@ -2,7 +2,8 @@
   <PropertyGroup>
     <TargetFramework>net5.0</TargetFramework>
     <GenerateAssemblyInfo>true</GenerateAssemblyInfo>
-    <LangVersion>8.0</LangVersion>
+    <LangVersion>9.0</LangVersion>
```

Comment:
Probably the most contentious change in this PR.  Bumping to C# 9 allows us to have nullable unconstrained generics like `T Foo<T>() { return default; }`.  A nice feature, but I don't know what the team's policy is on bumping language versions.  This branch does include a bump to .NET 5, so maybe it is a good place to also start bumping to C# 9?  For this PR I just bumped it in a couple places that were causing warnings, but it could easily be done project wide in one shot.

Reply:
this should be bumped at once for all the solution. We have four solutions:
Nethermind.sln
Baseline.sln
NDM.sln
EthereumTests.sln

and two submodules:
Int256
RocksDbSharp

has to be done and tested all at once after we merge pruning_uat with master and transition to 5.0 entirely